sentry-ruby/spec/sentry/transport/http_transport_rate_limiting_spec.rb

```diff
@@ -0,0 +1,252 @@
+require 'spec_helper'
+
+RSpec.describe "rate limiting" do
+  let(:configuration) do
+    Sentry::Configuration.new.tap do |config|
+      config.dsn = 'http://12345@sentry.localdomain/sentry/42'
+    end
+  end
+  let(:client) { Sentry::Client.new(configuration) }
+  let(:event) { client.event_from_message("foobarbaz") }
+  let(:data) do
+    subject.encode(event.to_hash)
+  end
+
+  subject { Sentry::HTTPTransport.new(configuration) }
+
+  describe "#is_rate_limited?" do
+    let(:transaction_event) do
+      client.event_from_transaction(Sentry::Transaction.new)
+    end
+
+    context "with only category limits" do
+      it "returns true for still limited category" do
+        subject.rate_limits.merge!("error" => Time.now + 60, "transaction" => Time.now + 60)
+
+        expect(subject.is_rate_limited?("event")).to eq(true)
+        expect(subject.is_rate_limited?("transaction")).to eq(true)
+      end
+
+      it "returns false for passed limited category" do
+        subject.rate_limits.merge!("error" => Time.now - 10, "transaction" => Time.now - 10)
+
+        expect(subject.is_rate_limited?("event")).to eq(false)
+        expect(subject.is_rate_limited?("transaction")).to eq(false)
+      end
+
+      it "returns false for not listed category" do
+        subject.rate_limits.merge!("transaction" => Time.now + 10)
+
+        expect(subject.is_rate_limited?("event")).to eq(false)
+      end
+    end
+
+    context "with only universal limits" do
+      it "returns true when still limited" do
+        subject.rate_limits.merge!(nil => Time.now + 60)
+
+        expect(subject.is_rate_limited?("event")).to eq(true)
+      end
+
+      it "returns false when passed limit" do
+        subject.rate_limits.merge!(nil => Time.now - 10)
+
+        expect(subject.is_rate_limited?("event")).to eq(false)
+      end
+    end
+
+    context "with both category-based and universal limits" do
+      it "prioritizes category limits" do
```

Comment:
The assertion and behavior in this test are correct, but the assumption is not. This links together with the comment above: Instead of _prioritizing_ the category limit, `is_rate_limited` needs to check _both_.

Can you add another test case like this?

```ruby
      it "always checks universal limits" do
        subject.rate_limits.merge!(
          "error" => Time.now - 10,
          nil => Time.now + 60
        )

        expect(subject.is_rate_limited?("event")).to eq(true)
      end
```